src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsassessforearlyreleaseapi/service/OffenderService.kt

```diff
@@ -61,7 +67,16 @@ class OffenderService(
       hdced = prisoner.homeDetentionCurfewEligibilityDate!!,
       crd = prisoner.conditionalReleaseDate,
     )
-    offender.assessments.add(Assessment(offender = offender, policyVersion = PolicyService.CURRENT_POLICY_VERSION.code))
+
+    var assessment = Assessment(offender = offender, policyVersion = PolicyService.CURRENT_POLICY_VERSION.code)
+    val offenderManager = probationService.getCurrentResponsibleOfficer(prisoner.prisonerNumber)
+    if (offenderManager != null) {
+      val responsibleCom = staffRepository.findByStaffIdentifier(offenderManager.id)
+        ?: createCommunityOffenderManager(offenderManager)
+      assessment = assessment.copy(responsibleCom = responsibleCom)
+    }
```

Comment:
**nit: (non-blocking)**: could also be expressed:

```
val responsibleCom = probationService.getCurrentResponsibleOfficer(prisoner.prisonerNumber)?.let {
    staffRepository.findByStaffIdentifier(it.id) ?: createCommunityOffenderManager(it)
}
val assessment = Assessment(offender = offender, policyVersion = PolicyService.CURRENT_POLICY_VERSION.code, responsibleCom = responsibleCom)
```

Reply:
Much better, I've added a TODO so that I do this on my next PR.